src/core/core-impl.hpp

```diff
@@ -58,9 +58,14 @@ class compositor_core_impl_t : public compositor_core_t
 
     wlr_seat *get_current_seat() override;
     uint32_t get_keyboard_modifiers() override;
+    void warp_cursor(wf::pointf_t pos) override;
+    /**
+     * The functions to hide and set the cursor use reference counting.
+     * A call to hide must come before a matched call to set.
+     */
     void set_cursor(std::string name) override;
     void hide_cursor() override;
-    void warp_cursor(wf::pointf_t pos) override;
+    int hide_ref_counter = 0;
```

Comment:
this counter belongs in wf::cursor_t

Reply:
I also thought this at first but the trouble was that init_xcursor() keeps getting called which keeps resetting the variable and it does not work as intended.

Reply:
I don't understand, why do you need to reset that variable in init_xcursor?